core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/federation/SilentIteration.java

```
@@ -34,9 +39,20 @@ protected BindingSet getNextElement() throws QueryEvaluationException {
 				return iter.next();
 		} catch (Exception e) {
 			// suppress
+			if (logger.isTraceEnabled()) {
+				logger.trace("Suppressed error in SILENT SERVICE: " + e.getMessage(), e);
+			}
 		}
 
 		return null;
 	}
 
+	@Override
+	protected void handleClose() throws QueryEvaluationException {
+		try {
+			Iterations.closeCloseable(iter);
+		} finally {
+			super.handleClose();
+		}
```

Should maybe this also fail silently?